### PR TITLE
added 'average' property to dsr target, added checkbox to dsr target cre...

### PR DESCRIPTION
**Pull request by susanm74**
...ate target form, added 'average or sum' column to dsr target list, and updated tests

### CODE REVIEW

File: grails-app/services/org/chai/kevin/dsr/DsrService.java [hunk 1]
```diff
@@ -96,11 +97,18 @@ private Value getDsrValue(DataElement dataElement, DataLocation dataLocation, Pe
...
+		if(target.getAverage()){
+			SumValue calculationValue = (SumValue) valueService.getCalculationValue(calculation, location, period, types);
+			if(calculationValue != null)
+				value = calculationValue.getAverage();
+		}
+		else{
+			CalculationValue<?> calculationValue = valueService.getCalculationValue(calculation, location, period, types);
+			if(calculationValue != null)
+				value = calculationValue.getValue();
+		}
```
Comment (fterrier) [inline grails-app/services/org/chai/kevin/dsr/DsrService.java:106]: aren't those 2 cases exactly the same ?

Comment (susanm74) [inline grails-app/services/org/chai/kevin/dsr/DsrService.java:106]: umm, one is getting the value, the other the average?

Comment (fterrier) [inline grails-app/services/org/chai/kevin/dsr/DsrService.java:106]: I see .. I guess moving the if deeper down would make things more readable .. like this maybe ?

SumValue calculationValue = (SumValue) valueService.getCalculationValue(calculation, location, period, types);
if(calculationValue != null && target.getAverage()) value = calculationValue.getAverage();
else if (calculationValue != null) value = calculationValue.getRatio();

**Comment by fterrier** [comment]
otherwise go ..

File: grails-app/services/org/chai/kevin/dsr/DsrService.java (hunk 1 above)
Comment (susanm74) [inline grails-app/services/org/chai/kevin/dsr/DsrService.java:106]: yeah, actually, both cases are returning a sumvalue, so i think i can combine, i updated it to almost exactly that, verifying now that it works, and will update.

SumValue calculationValue = (SumValue) valueService.getCalculationValue(calculation, location, period, types);
        if(calculationValue != null){
            if(target.getAverage()) value = calculationValue.getAverage();
            else value = calculationValue.getValue();
        }

**Pull request merged**